### DISCUSSION TITLE
How pre-boot DMA protection is detected?

### DISCUSSION THREAD

**Discussion by mhalano** [fwupd/fwupd#6194] [Q&A] [answered]
**Describe the question**
I would like to know how the security feature of Pre-Boot DMA protection is detected in the system so I can enable. I have a Dell laptop that doesn't have the option in the BIOS, so I would like to know why the command `fwupdmgr security` says it's `disabled` when should be `not-found`.

**fwupd version information**
`1.9.5

```shell
➜ fwupdmgr --version 
compile   org.freedesktop.fwupd         1.9.5
compile   com.hughsie.libxmlb           0.3.10
compile   com.hughsie.libjcat           0.1.9
runtime   org.freedesktop.fwupd-efi     1.4
compile   org.freedesktop.gusb          0.4.5
runtime   com.dell.libsmbios            2.4
runtime   org.freedesktop.gusb          0.4.5
runtime   org.kernel                    6.6.0-060600rc1-generic
runtime   org.freedesktop.fwupd         1.9.5
```

**Comment by hughsie** [comment, marked as answer]
For AMD it's the IVRS table https://github.com/fwupd/fwupd/blob/main/plugins/acpi-ivrs/fu-acpi-ivrs-plugin.c and for Intel is the DMAR table: https://github.com/fwupd/fwupd/blob/main/plugins/acpi-dmar/fu-acpi-dmar-plugin.c

**Comment by superm1** [comment]
To add to Richard's comments, some OEMs enable this feature only when certain other (seemingly) unrelated security options are enabled.  But it definitely requires firmware support to enable it.

**Reply by mhalano** [reply]
After some time I found out. Sorry for the absence of feedback. I had to enable Secure Boot on BIOS (I have a Dell machine). In some models there is an exclusively option, but not in my laptop.
